### PR TITLE
tailor: try to squash toad clothing bug

### PR DESCRIPTION
**Pull request by ab9rf**
this adds a test for an unmapped clothing size which will at least stop the making of toad-sized clothing -see #2916. a diagnostic is issued when this happens as it is a bug

### CODE REVIEW

**Review by myk002 — COMMENTED** [review]
need to add adamantine to:
- `set_material_order()`
- `tailor_setMaterialPreferences()`
- `setMaterials()` (in `plugins/lua/tailor.lua`)
- docs (in `docs/plugins/tailor.rst`)

File: plugins/tailor.cpp [hunk 1]
```diff
@@ -698,6 +712,7 @@ static void tailor_setMaterialPreferences(color_ostream &out, int32_t silkIdx,
     set_config_val(config, CONFIG_CLOTH_IDX, clothIdx - 1);
     set_config_val(config, CONFIG_YARN_IDX, yarnIdx - 1);
     set_config_val(config, CONFIG_LEATHER_IDX, leatherIdx - 1);
+    set_config_val(config, CONFIG_ADAMANTINE_IDX, leatherIdx - 1);
```
Comment (myk002) [inline plugins/tailor.cpp:715]:
```suggestion
    set_config_val(config, CONFIG_ADAMANTINE_IDX, adamantineIdx - 1);
```

Comment (myk002) [reply plugins/tailor.cpp:715]: function needs a new `int32_t adamantineIdx` param.

Comment (ab9rf) [reply plugins/tailor.cpp:715]: this is what i get for coding while still on my first cup of coffee

**Review by myk002 — APPROVED** [review]

**Pull request merged**